### PR TITLE
Consolidate development scripts into `Makefile`

### PR DESCRIPTION
**Pull request by evanrelf**
I noticed some duplication between commands defined in the `Makefile` and scripts added to the Nix shell's `PATH`. I think the `Makefile` is a better place for this stuff, so I consolidated everything there.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,9 @@
+ghcid: dev
+dev: ## Start ghcid
+	@ghcid --target lib:matchmaker --allow-eval --warnings -o ghcid.text
```
Comment (Kleidukos) [inline Makefile]: What is the difference between `-o ghcid.text | tail -f` and simply running ghcid on stdout?

Comment (evanrelf) [reply Makefile]: I'm not sure what the `-o ghcid.text` bit is doing here. I just left it as is assuming someone else wanted it that way.

Comment (Kleidukos) [reply Makefile]: hmm you're right, actually I think it was @JonathanLorimer who introduced it initially

Comment (JonathanLorimer) [reply Makefile]: it just outputs ghcid to a file. I have always used it, but not sure what the advantage is.

Comment (Kleidukos) [reply Makefile]: Okay, then I vote that we remove it for the time being. We can introduce a `$GHCID_OPTS` variable if it's really needed during day-to-day development.

Comment (evanrelf) [reply Makefile]: Okay I'll remove it

**Review by Kleidukos — APPROVED** [review]
Thank you :green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: Makefile (hunk 1 above)
Comment (Kleidukos) [reply Makefile]: Great, don't forget to squash your commits when merging. :)

**Pull request merged**